packages/@aws-cdk/cloudformation-include/test/valid-templates.test.ts

```diff
@@ -192,6 +192,30 @@ describe('CDK Include', () => {
     );
   });
 
+  test('can ingest a template with Fn::Sub in string form and output it unchanged', () => {
+    includeTestTemplate(stack, 'fn-sub-valid-string.json');
+
+    expect(stack).toMatchTemplate(
+      loadTestFileToJsObject('fn-sub-valid-string.json'),
+    );
+  });
+
+  test('can ingest a template with Fn::Sub in map form and output it unchanged', () => {
+    includeTestTemplate(stack, 'fn-sub-valid-map.json');
+
+    expect(stack).toMatchTemplate(
+      loadTestFileToJsObject('fn-sub-valid-map.json'),
+    );
+  });
+
+  test('can ingest a template with Fn::Sub with a key that is a logicalID and a literal and output it unchanged', () => {
+    includeTestTemplate(stack, 'fn-sub-shadow.json');
+
+    expect(stack).toMatchTemplate(
+      loadTestFileToJsObject('fn-sub-shadow.json'),
+    );
+  });
+
```

Comment:
Please also add a test that renames the logical ID of a resource referenced in Fn::Sub, and verifies the string is changed accordingly.